Release: Olyzaps upgrades, GTM tag update, Arbitrum Switcher, miscellaneous UI

nabioz opened the pull request.

nabioz committed:
Olyzaps upgrades (#822)

* UI tweaks

* Disable settings button

Commit:
feature: goBack on clicking away from bond ohm-modal (#814)

* feature: goBack on clicking away from bond ohm-modal

* prettier

brightiron committed:
refactoring stake view to introduce stateless metric component (#819)

cptnanon committed:
added event label to staking in GA (#827)

brightiron committed:
Moving Treasury Top Level Metrics to Shared Metric Component (#823)

* adding support for shared Metric Component to Treasury Dashboard

* Swapping false/true so it logically reads correctly.

cptnanon committed:
Added general olypmus GTM tag (#836)

* added general olypmus GTM tag

* refactor

appleseed-iii committed:
arbitrum buttons adding other chain (#832)

* arbitrum buttons adding other chain

* add migrator & gohm address, need wsohm

* add wsohm

* Arbitrum wrapping & catching network related errors

1. catch network related errors associated with non-existing contract addresses
2. redirect views that don't exist on a given network

* remove extra logs, remove warnings on prod

drondin committed:
Switch Subgraph to Protocol Metrics specific (#838)

biwano committed:
Added Vietnamese language (#834)

Co-authored-by: bilponse <bilponse@localhost.localdomain>

0xLienid committed:
Increasing z-index of tooltips so they are not hidden behind modals (#829)

* Increasing z-index of tooltips so they are not hidden behind modals

* merge conflict notes for translations

Co-authored-by: appleseed-iii <appleseed-iii@protonmail.com>

appleseed-iii commented on src/App.tsx:
```diff
@@ -128,9 +128,12 @@ function App() {
-      bonds.map(bond => {
-        dispatch(calcBondDetails({ bond, value: "", provider: loadProvider, networkID: networkId }));
-      });
+      // NOTE (appleseed) - tech debt - better network filtering for active bonds
+      if (networkId === 1 || networkId === 4) {
+        bonds.map(bond => {
+          dispatch(calcBondDetails({ bond, value: "", provider: loadProvider, networkID: networkId }));
+        });
+      }
```
trying to prevent as many failing contract calls as possible.

appleseed-iii commented on src/constants.ts:
```diff
@@ -96,8 +96,9 @@ export const addresses: IAddresses = {
     BONDINGCALC_ADDRESS: "0xcaaa6a2d4b26067a391e7b7d65c16bb2d5fa571a",
     CIRCULATING_SUPPLY_ADDRESS: "0x0efff9199aa1ac3c3e34e957567c1be8bf295034",
     TREASURY_ADDRESS: "0x31f8cc382c9898b273eff4e0b7626a6987c846e8",
-    // TODO (appleseed-lusd): swap this out
```
all these existing arbitrum contract addresses were added previously. I think they're not valid & could be removed?

appleseed-iii commented on src/hooks/web3Context.tsx:
```diff
@@ -178,18 +178,5 @@ export const Web3ContextProvider: React.FC<{ children: ReactElement }> = ({ chil
     ],
   );
 
-  useEffect(() => {
-    // logs non-functioning nodes && returns an array of working mainnet nodes
-    NodeHelper.checkAllNodesStatus(store.getState().network.networkId).then((validNodes: any) => {
-      validNodes = validNodes.filter((url: boolean | string) => url !== false);
-      if (!validNodes.includes(store.getState().network.uri) && NodeHelper.retryOnInvalid()) {
-        // force new provider...
-        setTimeout(() => {
-          window.location.reload();
-        }, 1);
-      }
-    });
-  }, []);
-
```
this is overkill now that we no longer use our own node

appleseed-iii commented on src/views/Bond/Bond.tsx:
```diff
@@ -43,6 +45,13 @@ const Bond = ({ bond }: { bond: IAllBondData }) => {
     return setSlippage(Number(e.target.value));
   };
 
+  const onClickAway = (): void => {
+    history.goBack();
+  };
+
+  const onClickModal = (e: any): void => {
+    e.stopPropagation();
+  };
```
allows you to click outside the bond modal to dismiss it

appleseed-iii commented on src/views/Zap/Zap.jsx:
```diff
@@ -7,9 +7,14 @@ import ZapStakeAction from "./ZapStakeAction";
 
 function Zap() {
   const { address, connect } = useWeb3Context();
+  const history = useHistory();
+  const networkId = useAppSelector(state => state.network.networkId);
+  usePathForNetwork({ pathName: "zap", networkID: networkId, history });
```
this is how we can handle network based redirects in the least amount of lines I could imagine today. this rather than putting a useEffect on each file.

appleseed-iii committed:
update exit language per sherpas (#849)

appleseed-iii committed:
1. remove outdated geoapify api key note from .env.example (#830)

2. shift padding on TreasuryDashboard Charts.
  A. Some charts had extra bottom padding
  B. Add Left padding. All charts had right padding without corresponding left padding
3. remove old Environment comments
4. remove old TreasuryDashboard.jsx, which was replaced by TreasuryDashboard.tsx previously
5. remove unused import from Graph.js
